clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/BaseIntegrationSpec.scala

```diff
@@ -80,6 +89,81 @@ abstract class BaseIntegrationSpec(clioDescription: String)
     */
   def elasticsearchUri: Uri
 
+  /** URL of vault server to use when getting bearer tokens for service accounts. */
+  private val vaultUrl = "https://clotho.broadinstitute.org:8200/"
+
+  /** Path in vault to the service account JSON to use in testing. */
+  private val vaultPath = "secret/dsde/gotc/test/clio/clio-account.json"
+
+  /** List of possible token-file locations, in order of preference. */
+  private val vaultTokenFiles = Seq(
+    new File("/etc/vault-token-dsde"),
+    new File(s"${System.getProperty("user.home")}/.vault-token")
+  )
+
+  /** Scopes needed from Google to write to our test-storage bucket. */
+  private val testStorageScopes = Seq(
+    "https://www.googleapis.com/auth/devstorage.read_write"
+  )
+
+  /**
+    * Service account credentials for use when accessing cloud resources.
+    */
+  protected lazy val serviceAccount: ServiceAccount = {
+    val vaultToken: String = vaultTokenFiles
+      .find(_.exists)
+      .map { file =>
+        new String(Files.readAllBytes(file.toPath)).stripLineEnd
+      }
+      .getOrElse {
+        sys.error("Vault token not found on filesystem!")
+      }
+
+    val vaultConfig = new VaultConfig()
+      .address(vaultUrl)
+      .token(vaultToken)
+      .build()
+
+    val vaultDriver = new Vault(vaultConfig)
+    val accountJSON =
+      vaultDriver
+        .logical()
+        .read(vaultPath)
+        .getData
+        .asScala
+        .toMap[String, String]
+        .asJson
+
+    accountJSON
+      .as[ServiceAccount]
+      .fold({ err =>
+        throw new RuntimeException(
+          s"Failed to decode service account JSON from Vault at $vaultPath",
+          err
+        )
+      }, identity)
+  }
+
+  /**
+    * Path to the cloud directory to which all GCP test data
+    * should be written.
+    */
+  lazy val rootTestStorageDir: Path = {
+    val storageOptions = StorageOptions
+      .newBuilder()
+      .setProjectId("broad-gotc-test-storage")
+      .setCredentials(serviceAccount.credentialForScopes(testStorageScopes))
+      .build()
+
+    val gcs: FileSystem = CloudStorageFileSystem.forBucket(
+      "broad-gotc-clio-test",
```

Comment:
I thought that the bucket format is `gs://broad-gotc-${ENV}-clio`? Is this named differently for some reason?

If it's different it should be documented in confluence, if it's the same it should share code with `EnvIntegrationSpec.rootPersistenceDir`

Reply:
It's named differently because that's how Henry made it. I'll ask if it'd be a big pain to rename.